src/Core/Datatypes/MetadataObject.h

```diff
@@ -0,0 +1,54 @@
+/*
+   For more information, please see: http://software.sci.utah.edu
+
+   The MIT License
+
+   Copyright (c) 2020 Scientific Computing and Imaging Institute,
+   University of Utah.
+
+   Permission is hereby granted, free of charge, to any person obtaining a
+   copy of this software and associated documentation files (the "Software"),
+   to deal in the Software without restriction, including without limitation
+   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+   and/or sell copies of the Software, and to permit persons to whom the
+   Software is furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included
+   in all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+   OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+   THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+   DEALINGS IN THE SOFTWARE.
+*/
+
+
+#ifndef CORE_DATATYPES_METADATAOBJECT_H
+#define CORE_DATATYPES_METADATAOBJECT_H
+
+#include <Core/Datatypes/Datatype.h>
+#include <Core/Datatypes/share.h>
+
+namespace SCIRun {
+namespace Core {
+namespace Datatypes {
+
+  class SCISHARE MetadataObject : public Datatype
+  {
+  public:
+    explicit MetadataObject(const std::string& programData);
+    std::string program() const { return programData_; }
```

Comment:
Just a nit-picky thing, you can also return a reference to `programData_` like:

`const std::string& program() { return programData_; }

Reply:
Yeah I usually return strings like that, I guess I was typing too quickly here

Reply:
Also, I was thinking the string return type would be a placeholder for whatever this class was going to do (it’s still prototype), so I’ll probably move that method soon anyway.